File: virtual-world-simulator/Wolf.cpp
#include "Wolf.h"

Wolf::Wolf(World* world, Point position) :
	Animal{ world, position, 'W', 9, 5 }
{
}

bool Wolf::collision(Entity& entity)
{
	Wolf* wolf = dynamic_cast<Wolf*>(&entity);
	if (wolf != nullptr)
	{
		this->breed<Wolf>(*wolf);
		return false;
	}

	return Animal::collision(entity);
}

File: Java/virtual-world-simulator/src/Entities/Animals/Wolf.java
package Entities.Animals;

import App.World;
import Common.Point;
import Entities.Entity;

import javax.swing.*;

public class Wolf extends Animal
{
    public Wolf(World world, Point position)
    {
        super(new ImageIcon("assets/wolf.png"), world, position, 'W', 9, 5);
    }

    public boolean Collision(Entity entity)
    {
        if(entity instanceof Wolf)
        {
            Wolf wolf = new Wolf(this.world, new Point(-1, -1));
            AddChild(wolf);
            return false;
        }

        return super.Collision(entity);
    }
}
